HHS/Head-Start-TTADP
# [TTAHUB-3361] ssdi refactor

## Description of change

The pull request refactors the SSDI by restructuring existing logic, updating API responses, and optimizing query handling to reduce redundancy. It introduces more streamlined functions, improves error handling, and enhances the integration of new feature flags for toggling specific behavior in the application.

## How to test

1. Verify API Endpoints: Check that all SSDI-related API endpoints return expected results with the correct data structure.
2. Feature Flags: Toggle feature flags and ensure they activate the appropriate behavior.
3. Error Handling: Introduce various invalid inputs to confirm the improved error handling is functioning correctly.

List:
- http://192.168.1.95:3000/api/ssdi/api/list
- http://192.168.1.95:3000/api/ssdi/dataRequests/list

Filters:
- http://192.168.1.95:3000/api/ssdi/api/dashboards/qa/no-tta.sql/filters
- http://192.168.1.95:3000/api/ssdi//dataRequests/user/communication-logs.sql/filters

Run:
- http://192.168.1.95:3000/api/ssdi/api/dashboards/qa/no-tta.sql
- http://192.168.1.95:3000/api/ssdi//dataRequests/user/communication-logs.sql

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3361


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
